### DISCUSSION TITLE
how to get the ip and port from fastdds discover module.

### DISCUSSION THREAD

**Discussion by fengmao31** [eProsima/Fast-DDS#3417] [Q&A]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

I want to use fastdds to dicover the node in the network and use other trasnport method. I want to use the ip and port from fastdds discover module. 

### Current behavior

I read the document and search online, read some related project, but I cannot find the api.

### Steps to reproduce

 I want to export the ip and port from fastdds discover module. 

### Fast DDS version/commit

2.9
1.5

### Platform/Architecture

Ubuntu Focal 20.04 amd64

### Transport layer

UDPv4

**Comment by jsan-rt** [comment]
Hi @fengmao31 ,

You can get the IP and ports of each discovered entity via the use of discovery callbacks. You can check [this link](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/disc_callbacks.html) for more information regarding the specifics of the implementation of a custom listener. The discovery info passed to the discovery callbacks contains, among other things the locator information (protocol, IP and port) for that entity.

I'm moving this conversation to Discussions since it's not an issue.

**Reply by fengmao31** [reply]
> Hi @fengmao31 ,
> 
> You can get the IP and ports of each discovered entity via the use of discovery callbacks. You can check [this link](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/disc_callbacks.html) for more information regarding the specifics of the implementation of a custom listener. The discovery info passed to the discovery callbacks contains, among other things the locator information (protocol, IP and port) for that entity.
> 
> I'm moving this conversation to Discussions since it's not an issue.

I know the callback api. but how to get the value of  ip and port in the domainParticipants, writer and reader.
